installer/pkg/components/blobserve/deployment.go

```diff
@@ -18,55 +18,25 @@ import (
 func deployment(ctx *common.RenderContext) ([]runtime.Object, error) {
 	labels := common.DefaultLabels(Component)
 
+	volumeName := "pull-secret"
+	var secretName string
+	if pointer.BoolDeref(ctx.Config.ContainerRegistry.InCluster, false) {
+		secretName = dockerregistry.BuiltInRegistryAuth
+	} else {
```

Comment:
```suggestion
	} else if ctx.Config.ContainerRegistry.External != nil {
```

Reply:
Ok. I did this on the basis that it was either/or (protected by validation), but not against defensive coding